src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -558,6 +558,9 @@ AdaptiveExecutorCreateScan(CustomScan *scan)
 	scanState->customScanState.methods = &AdaptiveExecutorCustomExecMethods;
 	scanState->PreExecScan = &CitusPreExecScan;
 
+	scanState->finishedPreScan = false;
+	scanState->finishedRemoteScan = false;
```

Comment:
Was thrown off by this a bit as it seemed unrelated to the boolean we are adding here. But after expanding the context I saw we `palloc0`, so the value was implicitly `false` already. I do like the verbosity here though, so thanks for adding.